avocado/utils/network/interfaces.py

```diff
@@ -612,6 +612,52 @@ def is_bond(self):
             LOG.debug(msg)
             return False
 
+    def is_veth(self):
+        """Check if interface is a Virtual Ethernet.
+
+        This method checks if the interface is a Virtual Ethernet or not.
+
+        rtype: bool
+        """
+        if not os.path.isfile(f"/sys/class/net/{self.name}/device/devspec"):
+            raise ValueError("Please provide valid inerface name")
+        cmd = f"cat /sys/class/net/{self.name}/device/devspec"
+        output = run_command(cmd, self.host)
+        if "l-lan" in output:
+            return True
+        return False
+
+    def is_vnic(self):
+        """Check if interface is a virtual network.
+
+        This method checks if the interface is a virtual NIC or not.
+
+        rtype: bool
+        """
+        if not os.path.isfile(f"/sys/class/net/{self.name}/device/devspec"):
+            raise ValueError("Please provide valid inerface name")
+        cmd = f"cat /sys/class/net/{self.name}/device/devspec"
+        output = run_command(cmd, self.host)
+        if "vnic" in output:
+            return True
+        return False
```

Comment:
can utility  is_vnic and is_veth combined name can be like 

```
is_virtualnet(self, mode=''):
     if mode='vnic':
        check_string= 'vnic'
     elif mode='veth':
          check_string= '"l-lan'
     else:
           raise NWException("Not supported Format")
  
   if not os.path.isfile(f"/sys/class/net/{self.name}/device/devspec"):
            raise NWException ("Please provide valid inerface name")
        cmd = f"cat /sys/class/net/{self.name}/device/devspec"
        output = run_command(cmd, self.host)
        if checkstring  in output:
            return True
        return False
  ```

Reply:
utility is only to know mode.. if it is already known.. than no need to check fro if it is vnic .. you got the point ?

Reply:
with suggested way.. it is like you are passing vnic and returning vnic

Reply:
yes reasonable just looking to minimize the code uses may be common part can be push in another module